fe/fe-core/src/main/java/com/starrocks/catalog/GlobalFunctionMgr.java

```diff
@@ -109,12 +111,27 @@ private void addFunction(Function function, boolean isReplay, boolean allowExist
      * @return a new list of functions with the given function added or replaced.
      */
     public static ImmutableList<Function> addOrReplaceFunction(Function function, List<Function> existFuncs) {
-        return ImmutableList.<Function>builder()
-                .addAll(existFuncs.stream()
-                        .filter(f -> !function.compare(f, Function.CompareMode.IS_IDENTICAL))
-                        .collect(ImmutableList.toImmutableList()))
-                .add(function)
-                .build();
+        List<Function> filteredFuncs = existFuncs.stream()
+                .filter(f -> !function.compare(f, Function.CompareMode.IS_IDENTICAL))
+                .collect(Collectors.toList());
+
+        filteredFuncs.add(function);
+
+        filteredFuncs.sort((f1, f2) -> {
```

Comment:
If the parameter types are float and Int, how does this match?

Reply:
According to the order in which UDFs are added

Reply:
That doesn't meet expectations either?

Reply:
I will handle it in the next PR